pyteal/ast/router.py

```diff
@@ -459,34 +560,92 @@ def wrap_handler(
                     abi.MethodReturn(output_temp),
                     Approve(),
                 )
+        else:
+            subroutine_ify = ASTBuilder.__de_abify_subroutine(handler)
+            return Seq(
+                subroutine_ify(),
+                Approve(),
+            )
+
+    @staticmethod
+    def __de_abify_subroutine(
```

Comment:
I think since this logic is meant for `frame_pointers` usage, it would help to evoke this by adding `fp` or `frame_pointers` to the method name. Also, out of symmetry, it's probably preferable to summarize the non-frame pointers code generation in lines 527-562 with a function as well.

Reply:
Once that is done, there might be other "obvious" points where the logic can be unified.

Reply:
or it may be the case that you can pass `use_frame_pointers` into this function and use in both places.

Reply:
The suggestion on renaming the method name to be related with frame pointers makes a lot of sense.

I prefer packing vanilla flavor of code generation to be another function, rather controlled by passing `use_frame_pointer`, for that will be way too complex and hard to maintain the resulting code.